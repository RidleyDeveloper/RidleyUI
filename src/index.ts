export * from "./components/Example";
// export other wrapped components as you add them
