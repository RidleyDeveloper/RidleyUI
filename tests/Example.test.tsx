import renderer from "react-test-renderer";
import { describe, expect, it } from "vitest";
import { Button, ExampleCard, RidleyAlert, RidleyBadge } from "../src";

describe("Button", () => {
	it("renders default button correctly", () => {
		const tree = renderer.create(<Button>Default Button</Button>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("renders ridley variant correctly", () => {
		const tree = renderer
			.create(<Button variant="ridley">Ridley Button</Button>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("renders ridley-white variant correctly", () => {
		const tree = renderer
			.create(<Button variant="ridley-white">Ridley White Button</Button>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});

describe("ExampleCard", () => {
	it("renders card correctly", () => {
		const tree = renderer.create(<ExampleCard />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});

describe("RidleyBadge", () => {
	it("renders badge correctly", () => {
		const tree = renderer
			.create(<RidleyBadge>Test Badge</RidleyBadge>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});

describe("RidleyAlert", () => {
	it("renders alert correctly", () => {
		const tree = renderer
			.create(<RidleyAlert>Test Alert</RidleyAlert>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
