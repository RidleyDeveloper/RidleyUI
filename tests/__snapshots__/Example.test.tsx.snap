// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders default button correctly 1`] = `
<button
  className="rui-focus btn btn-secondary"
  onClick={[Function]}
>
  Default Button
</button>
`;

exports[`Button > renders ridley variant correctly 1`] = `
<button
  className="rui-focus btn-ridley btn btn-secondary"
  onClick={[Function]}
>
  Ridley Button
</button>
`;

exports[`Button > renders ridley-white variant correctly 1`] = `
<button
  className="rui-focus btn-ridley-white btn btn-secondary"
  onClick={[Function]}
>
  Ridley White Button
</button>
`;

exports[`ExampleCard > renders card correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Ridley UI Card
    </h5>
    <p
      className="card-text"
    >
      This is an example card using ReactStrap with Ridley styling.
    </p>
    <div
      className="d-flex gap-2"
    >
      <button
        className="rui-focus btn-ridley btn btn-secondary"
        onClick={[Function]}
      >
        Primary Action
      </button>
      <button
        className="rui-focus btn-ridley-white btn btn-secondary"
        onClick={[Function]}
      >
        Secondary Action
      </button>
    </div>
  </div>
</div>
`;

exports[`RidleyAlert > renders alert correctly 1`] = `
<div
  className="alert alert-primary fade"
  role="alert"
>
  Test Alert
</div>
`;

exports[`RidleyBadge > renders badge correctly 1`] = `
<span
  className="badge bg-primary"
>
  Test Badge
</span>
`;
